Tests/FluentAssertions.Specs/Xml/XDocumentAssertionSpecs.cs

```diff
@@ -1231,6 +1231,221 @@ public void When_asserting_a_document_has_an_element_with_a_null_xname_it_should
         }
     }
 
+    public class HaveElementWithValue
+    {
+        [Fact]
+        public void Throws_when_null()
+        {
+            // Arrange
+            XDocument element = null;
+
+            // Act
+            Action act = () => element.Should().HaveElementWithValue("child", "b");
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*child*b*element itself is <null>*");
+        }
+
+        [Fact]
+        public void Has_element_with_specified_value()
```

Comment:
🔧This doesn't represent the behavior you expect.

Reply:
I don't get, what you mean. This test passes, when the document has both an element and a particular value for this element.

Reply:
These days, we try to write the test names as a fact, preferably without using `when` and `should` (like we did in the past).  In this case, I probably would have named it `The_expected_element_with_the_expected_value_is_valid`. But I'm nitpicking here.

Reply:
Ah ok, changed that :)